ui/raidboss/data/07-dt/raid/r4s.ts

```diff
@@ -1404,17 +1404,41 @@ const triggerSet: TriggerSet<Data> = {
           ? (data.midnightCardFirst ? output.cardinals!() : output.intercards!())
           : (data.midnightCardFirst ? output.intercards!() : output.cardinals!());
 
-        return output.combo!({ dir: dirStr, mech: firstMechStr });
+        const typeStr = data.midnightFirstAdds === 'wings' ? output.wings!() : output.guns!();
+
+        return output.combo!({ dir: dirStr, mech: firstMechStr, type: typeStr });
       },
       outputStrings: {
         combo: {
-          en: '${dir} => ${mech}',
-          de: '${dir} => ${mech}',
-          ja: '${dir} => ${mech}',
-          cn: '${dir} => ${mech}',
+          en: '${dir} => ${type} + ${mech}',
+          de: '${dir} => ${type} + ${mech}',
+          ja: '${dir} => ${type} + ${mech}',
+          cn: '${dir} => ${type} + ${mech}',
+        },
+        cardinals: {
+          en: 'Cardinals',
+          de: 'Kardinal',
+          fr: 'Cardinaux',
+          ja: '十字回避',
+          cn: '去正点',
+          ko: '십자방향으로',
+        },
+        intercards: {
+          en: 'Intercards',
+          de: 'Interkardinal',
+          fr: 'Intercardinal',
+          ja: '斜めへ',
+          cn: '去四角',
+          ko: '대각선 쪽으로',
```

Comment:
Was there a reason this wasn't sufficient or accurate?
```
        cardinals: Outputs.cardinals,
        intercards: Outputs.intercards,
```
If there's a translation issue with those, it should probably be fixed in `resources/outputs.ts`, rather than just hard-coding these values.

Reply:
In the `resources/outputs.ts` cardinals only for `十字` and intercards for `四角`. This translation does not specify whether the AoE are present at the place or whether the player is required to get the place. Besides, since this output may already be used in other triggers, So there is no new changes be made for `resources/outputs.ts`

Reply:
Ahh, okay, thank

> In the `resources/outputs.ts` cardinals only for `十字` and intercards for `四角`. This translation does not specify whether the AoE are present at the place or whether the player is required to get the place. Besides, since this output may already be used in other triggers, So there is no new changes be made for `resources/outputs.ts`

Ahh, okay, that makes sense.  Thanks for clarifying.  So just to check (and not questioning the translation, just want to make sure the intent of the original code/callout is clear):  For `cardinals`, the updated output string accurately conveys that cardinals are the safe spot, and you've added that there is an AoE that also needs to be avoided while the player is at the cardinal?   Or the updated string only conveys that there is an AoE at the cardinal that needs to be avoided (and might that cause the player to mistakenly then move to the intercard instead)?

Reply:
like the:
- `Cardinals => Avoid Line + Stack With Partner`: `去正点 => 躲避直线 + 与搭档分摊`

First `Cardinals` means you needs to go to cardinals, and `Avoid Line` means a line AoE from intercards is pending, `Stack With Partner` indicate you need to stack with partner.

- `Cardinals => Donut  + Spread`: `去正点 => 月环 + 分散`

First `Cardinals` means you needs to go to cardinals, and `Donut` means there is a Donut in the cardinals position, `Spread` indicate you need to spread.

The first only for safe spot location, the second for which type in the safe spot

In fact, the major changes for this pr is the size of the safe spot of line AoE and Donut is difference, the safe spot of Donut is smaller than Line AoE.

Reply:
Maybe 
- `Go Cardinals => Avoid Line + Stack With Partner`
- `Go Intercards => Donut  + Spread`

is better

Reply:
> Maybe
> 
> * `Go Cardinals => Avoid Line + Stack With Partner`
> * `Go Intercards => Donut  + Spread`
> 
> is better, which can represent this action

That makes sense to me (and thanks for the clarification, super helpful).  I don't have any other comments, other than some lingering hesitation over the callout being somewhat long and less concise than before.

If it's okay, maybe we can leave this PR up for another day or so to see if anyone else has feelings about that -- if not, I'll approve & merge.
(cc: @valarnin , @JLGarber )

Reply:
(I'll note here up-front that my static isn't here yet, so while I know how the mechanic works I haven't seen it firsthand.)

I think `Go Intercards + Donut + Spread` is good enough. Personally I would drop "Go", but that's opinion only. The rest of this one is as short as realistically possible without losing information.

`Go Cardinals => Avoid Line + Stack With Partner` might be shortened to `Cardinals + Avoid Line + Partner Stack` perhaps. As before, it's difficult to shorten this further. (I do understand the use of the => arrow separator here, but using only plus signs would make it clear that all three of these actions take place at the same time, instead of implying "go cardinals, then avoid the line and the partner stack".)

Those two notes aside, I think these outputs are as short as reasonably possible. They are long, but not unreasonably so.

Reply:
> I think `Go Intercards + Donut + Spread` is good enough. Personally I would drop "Go", but that's opinion only. The rest of this one is as short as realistically possible without losing information.
> 
> `Go Cardinals => Avoid Line + Stack With Partner` might be shortened to `Cardinals + Avoid Line + Partner Stack` perhaps. [...]

Those make sense to me.  I think that's probably as short as it can be without dropping actual info.